src/Microsoft.NET.Workloads.Vsman/Microsoft.NET.Workloads.Vsman.vsmanproj

```diff
@@ -0,0 +1,19 @@
+<Project Sdk="Microsoft.VisualStudio.Internal.MicroBuild.Vsman">
+
+  <PropertyGroup>
+    <TargetFramework>$(SdkTargetFramework)</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <MergeManifest Include="$(VsixOutputPath)**\*.json" />
+  </ItemGroup>
+
+  <!-- Required for setting the BuildVersion property which MicroBuild.Vsman uses to set the version of the component in the .vsman file. -->
+  <!-- See: https://devdiv.visualstudio.com/DevDiv/_wiki/wikis/DevDiv.wiki/29608/How-to-Build-a-vsman-file?anchor=creating-your-.vsman-file -->
+  <Target Name="GetBuildVersion">
+    <PropertyGroup>
+      <BuildVersion>$(WorkloadVersion)</BuildVersion>
```

Comment:
Might this cause collisions if it's being pushed to a drop somewhere for multiple consecutive builds of the same PR/workload version?

Reply:
This sets the version of the workload that is put into the vsman file that is then put into the VS insertion PR. This version value is extracted from the `.metadata` file from the drop and passed in as a property to this project build. The drop has a unique hash value that represents the drop, so even if the version number wasn't updated (for some reason) but the files were, they won't clash when the drop is uploaded.